demo/app/script.ts

```diff
@@ -4,26 +4,29 @@ import { nanoid } from 'nanoid';
 
 import { getRandomName } from './utils/fake-names';
 import { getSpaceNameFromUrl } from './utils/url';
-
 import Spaces from '../../src/Spaces';
 import { renderAvatars, renderSelfAvatar } from './components/avatar-stack';
 import { renderFeatureDisplay } from './components/feature-display';
 
-(async () => {
-  const clientId = nanoid();
-  const ably = new Ably.Realtime.Promise({ authUrl: `/api/ably-token-request?clientId=${clientId}`, clientId });
+const clientId = nanoid();
+
+const ably = new Ably.Realtime.Promise({ authUrl: `/api/ably-token-request?clientId=${clientId}`, clientId });
+
+const spaces = new Spaces(ably);
+
+export const space = spaces.get(getSpaceNameFromUrl(), { offlineTimeout: 60_000 });
+
+export const selfName = getRandomName();
```

Comment:
I'd prefer if exports were all defined at the bottom of the file, it's very easy to leave exports that should not be there in a file (and it disables the eslint/ts check for an unused variable too!)